addons/account/models/account_payment.py

```diff
@@ -193,6 +193,8 @@ def _onchange_journal(self):
                 active_ids = self._context.get('active_ids')
                 invoices = self.env['account.move'].browse(active_ids)
                 self.amount = abs(self._compute_payment_amount(invoices, self.currency_id, self.journal_id, self.payment_date))
+            if self.payment_type == 'inbound':
+                domain['partner_bank_account_id'] = [('partner_id', '=', self.company_id.partner_id.id)]
```

Comment:
also here use the invoice company

Reply:
These changes are not required anymore if we are using the invoice company.